File: bdp-devops/ambari/ambari-server/src/test/java/org/apache/ambari/server/state/action/JobTest.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.ambari.server.state.action;

import org.junit.Assert;
import org.junit.Test;

public class JobTest {

  private Action createNewJob(long id, String jobName, long startTime) {
    ActionId jId = new ActionId(id, new ActionType(jobName));
    Action job = new ActionImpl(jId, startTime);
    return job;
  }

  private Action getRunningJob(long id, String jobName, long startTime)
      throws Exception {
    Action job = createNewJob(id, jobName, startTime);
    verifyProgressUpdate(job, ++startTime);
    return job;
  }

  private Action getCompletedJob(long id, String jobName, long startTime,
      boolean failedJob) throws Exception {
    Action job = getRunningJob(1, "JobNameFoo", startTime);
    completeJob(job, failedJob, ++startTime);
    return job;
  }

  private void verifyNewJob(Action job, long startTime) {
    Assert.assertEquals(ActionState.INIT, job.getState());
    Assert.assertEquals(startTime, job.getStartTime());
  }


  @Test
  public void testNewJob() {
    long currentTime = System.currentTimeMillis();
    Action job = createNewJob(1, "JobNameFoo", currentTime);
    verifyNewJob(job, currentTime);
  }

  private void verifyProgressUpdate(Action job, long updateTime)
      throws Exception {
    ActionProgressUpdateEvent e = new ActionProgressUpdateEvent(job.getId(),
        updateTime);
    job.handleEvent(e);
    Assert.assertEquals(ActionState.IN_PROGRESS, job.getState());
    Assert.assertEquals(updateTime, job.getLastUpdateTime());
  }


  @Test
  public void testJobProgressUpdates() throws Exception {
    long currentTime = 1;
    Action job = createNewJob(1, "JobNameFoo", currentTime);
    verifyNewJob(job, currentTime);

    verifyProgressUpdate(job, ++currentTime);
    verifyProgressUpdate(job, ++currentTime);
    verifyProgressUpdate(job, ++currentTime);

  }

  private void completeJob(Action job, boolean failJob, long endTime)
      throws Exception {
    ActionEvent e = null;
    ActionState endState = null;
    if (failJob) {
      e = new ActionFailedEvent(job.getId(), endTime);
      endState = ActionState.FAILED;
    } else {
      e = new ActionCompletedEvent(job.getId(), endTime);
      endState = ActionState.COMPLETED;
    }
    job.handleEvent(e);
    Assert.assertEquals(endState, job.getState());
    Assert.assertEquals(endTime, job.getLastUpdateTime());
    Assert.assertEquals(endTime, job.getCompletionTime());
  }


  @Test
  public void testJobSuccessfulCompletion() throws Exception {
    long currentTime = 1;
    Action job = getRunningJob(1, "JobNameFoo", currentTime);
    completeJob(job, false, ++currentTime);
  }

  @Test
  public void testJobFailedCompletion() throws Exception {
    long currentTime = 1;
    Action job = getRunningJob(1, "JobNameFoo", currentTime);
    completeJob(job, true, ++currentTime);
  }

  @Test
  public void completeNewJob() throws Exception {
    long currentTime = 1;
    Action job = createNewJob(1, "JobNameFoo", currentTime);
    verifyNewJob(job, currentTime);
    completeJob(job, false, ++currentTime);
  }

  @Test
  public void failNewJob() throws Exception {
    long currentTime = 1;
    Action job = createNewJob(1, "JobNameFoo", currentTime);
    verifyNewJob(job, currentTime);
    completeJob(job, true, ++currentTime);
  }

  @Test
  public void reInitCompletedJob() throws Exception {
    Action job = getCompletedJob(1, "JobNameFoo", 1, false);
    ActionId jId = new ActionId(2, new ActionType("JobNameFoo"));
    ActionInitEvent e = new ActionInitEvent(jId, 100);
    job.handleEvent(e);
    Assert.assertEquals(ActionState.INIT, job.getState());
    Assert.assertEquals(100, job.getStartTime());
    Assert.assertEquals(-1, job.getLastUpdateTime());
    Assert.assertEquals(-1, job.getCompletionTime());
    Assert.assertEquals(2, job.getId().actionId);
  }


}

File: bdp-devops/ambari-versions/ambari-3.0.0.0-SNAPSHOT/contrib/views/pig/src/main/resources/ui/pig-web/test/unit/models/job_test.js
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


moduleForModel('job', 'App.Job',{
  needs:['model:file']
});

test('file relationship', function() {
  expect(2);
  var Job = this.store().modelFor('job'),
      relationship = Ember.get(Job, 'relationshipsByName').get('pigScript');

  equal(relationship.kind, 'belongsTo');
  equal(relationship.type.typeKey, App.File.typeKey);
});

test('duration format',function() {
  expect(3);
  var job = this.subject();

  Em.run(function(){
    job.set('duration', 60);
  });

  equal(job.get('durationTime'),"1 min, 0 sec");

  Em.run(function(){
    job.set('duration', 60*60);
  });

  equal(job.get('durationTime'),"1 hrs, 0 min, 0 sec");

  Em.run(function(){
    job.set('duration', 60*60*24);
  });

  equal(job.get('durationTime'),"24 hrs, 0 min, 0 sec");
});

test('percentStatus',function() {
  expect(2);
  var job = this.subject();

  Em.run(function(){
    job.setProperties({'percentComplete': 50,'status':'RUNNING'});
  });

  equal(job.get('percentStatus'),50);

  Em.run(function(){
    job.set('status','FAILED');
  });

  equal(job.get('percentStatus'),100);

});

test('setting argumets array property',function () {
  var job = this.subject(),
      args = job.get('argumentsArray');

  ok(args.length === 0 && !job.get('templetonArguments'));

  Em.run(function(){
    job.set('argumentsArray',args.pushObject('test_agr') && args);
  });

  ok(job.get('argumentsArray').length == 1 && job.get('templetonArguments') == 'test_agr')
});

